src/cdk/v2/destinations/algolia/procWorkflow.yaml

```diff
@@ -32,12 +39,33 @@ steps:
         filters: .properties.filters,
         objectIDs: .properties.objectIds,
         positions: .properties.positions,
+        value: $.isDefinedAndNotNull(.properties.currency) ? .properties.value : null,
+        currency: .properties.currency,
         userToken: {{{{$.getGenericPaths("userId", "||")}}}},
         eventName: event,
-        eventType: eventType
+        eventType: eventType,
+        eventSubtype: eventSubType
       });
       $.context.payload = $.genericpayloadValidator(payload);
 
+  - name: prepareObjectDataBlock
+    condition: $.context.payload.eventType === "conversion" && $.isDefinedAndNotNull(^.message.properties.products) && Array.isArray(^.message.properties.products)
+    description: |
+      Populate list of objectData
+    template: |
+      const populateObjectData = function(singleProduct) {
+      singleProduct.({
+          "queryID" :  $.isDefinedAndNotNull(.queryID) ? String(.queryID) : null,
+          "price": $.isDefinedAndNotNull(.price) && $.isDefinedAndNotNull(^.message.properties.currency) ? String(.price) : null,
+          "quantity": $.isDefinedAndNotNull(.quantity)? Number(.quantity) : null,
+          "discount": $.isDefinedAndNotNull(.discount) ? String(.discount) : null
+      })
+      };
+      const products = ^.message.properties.products
+      products.map(function(singleProduct) {
```

Comment:
you don't need to use map, you can write
```
products.({
          "queryID" :  $.isDefinedAndNotNull(.queryID) ? String(.queryID) : null,
          "price": $.isDefinedAndNotNull(.price) && $.isDefinedAndNotNull(^.message.properties.currency) ? String(.price) : null,
          "quantity": $.isDefinedAndNotNull(.quantity)? Number(.quantity) : null,
          "discount": $.isDefinedAndNotNull(.discount) ? String(.discount) : null
      })[]
```

Reply:
changed